packages/devtools_app/test/standalone_ui/vs_code/devtools_test.dart

```diff
@@ -52,31 +72,93 @@ void main() {
   }
 
   group('$DevToolsSidebarOptions', () {
-    testWidgetsWithWindowSize(
-      'includes DevTools screens',
-      windowSize,
-      (tester) async {
-        await pumpDevToolsSidebarOptions(tester);
-        expect(find.text('DevTools'), findsOneWidget);
-        for (final screen in ScreenMetaData.values) {
-          final include = DevToolsSidebarOptions.includeInSidebar(screen);
-          expect(
-            find.text(screen.title!),
-            include ? findsOneWidget : findsNothing,
+    for (final hasDebugSessions in [true, false]) {
+      final debugSessions = hasDebugSessions
+          ? {
+              'test session': generateDebugSession(
+                debuggerType: 'Flutter',
+                deviceId: 'macos',
+                flutterMode: 'debug',
+                projectRootPath: testDtdProjectRoot,
+              ),
+            }
+          : <String, EditorDebugSession>{};
+
+      testWidgetsWithWindowSize(
+        'pumps DevTools screens ${hasDebugSessions ? 'with' : 'without'} debug '
+        'sessions',
+        windowSize,
+        (tester) async {
+          await pumpDevToolsSidebarOptions(
+            tester,
+            debugSessions: debugSessions,
           );
-        }
-        expect(find.byIcon(Icons.open_in_browser_outlined), findsOneWidget);
-      },
-    );
 
-    testWidgetsWithWindowSize(
-      'includes DevTools extensions',
-      windowSize,
-      (tester) async {
-        await pumpDevToolsSidebarOptions(tester);
-        expect(find.text('DevTools Extensions'), findsOneWidget);
-        expect(find.text('bar'), findsOneWidget);
-      },
-    );
+          expect(find.text('DevTools'), findsOneWidget);
+          for (final screen in ScreenMetaData.values) {
+            final include = DevToolsSidebarOptions.includeInSidebar(screen);
+            // Do not check the 'simple' or 'home' screens because they do not
+            // have a title we can verify against.
+            if (screen != ScreenMetaData.simple &&
+                screen != ScreenMetaData.home) {
+              expect(
+                find.text(screen.title!),
+                include ? findsOneWidget : findsNothing,
+                reason: 'Unexpected find result for ${screen.id} screen.',
```

Comment:
nit: Having different text for each condition would make it faster to understand what the failure was (whether it was expected and not found, or expected and not found) without having to re-run the test.

Reply:
the test message will already say something like "Expected 0 widgets but found 1" or "Expected exactly 1 widget but found 0"